### ISSUE TITLE
Relationship to WebGPU

### ISSUE THREAD

**Issue by aboba** [w3c/mediacapture-transform#34]
Mediacapture-Transform's MSTProcessor interface uses `VideoFrame` . My understanding is that this represents a GPU buffer, but the relationship to [WebGPU's GPUBuffer](https://gpuweb.github.io/gpuweb/#gpu-device) is unclear to me. 

[WebGPU](https://gpuweb.github.io/gpuweb/), is going into [Origin Trial](https://groups.google.com/a/chromium.org/g/blink-dev/c/K4_egTNAvTs/m/ApS804L_AQAJ), so it would be useful to be able to try out using Mediacapture-Transform and WebGPU together (e.g. for "Funny Hats" scenarios).

**Comment by dogben** [comment]
There is quite a bit of discussion in https://github.com/gpuweb/gpuweb/issues/1380 about import.

@Kangz can confirm, but since [VideoFrame can now be created from a CanvasImageSource](https://github.com/w3c/webcodecs/issues/158), that covers WebGPU also.

**Comment by Kangz** [comment]
Yes that's correct. WebGPU can render to canvases so the `CanvasImageSource` path should work. Maybe additional paths are added in the future but that one should work (at least in terms of spec) today.

Note that `GPUBuffer` represents unstructured GPU memory (think `gpu_malloc`) while `GPUTexture` represents 2D (or 1D / 3D) memory with a specific format in an opaque layout. What we call "buffer" in the media world is closer in concept to a `GPUTexture`.

**Comment by alvestrand** [comment]
There's a demo of breakout box / WebGPU interaction at https://webrtc.github.io/samples/src/content/insertable-streams/webgpu/

**Comment by dontcallmedom** [comment]
related WebGPU integration discussions in https://github.com/webmachinelearning/webnn/issues/226#issuecomment-1010752362

The [sample above notes](https://github.com/webrtc/samples/blob/b4734bc531f36e0c350eca6c177ecf30911e2820/src/content/insertable-streams/webgpu/js/multi_video_main.js#L163) in particular:
```js
  // Using GPUExternalTexture(when it's implemented for Breakout Box frames) will
  // avoid making extra copies through ImageBitmap.
```

**Comment by aboba** [comment]
[WebCodecs PR 412](https://github.com/w3c/webcodecs/pull/412) proposes conversion of `VideoFrame` to `GPUExternalTexture`.
